src/gam/gapi/__init__.py

```diff
@@ -218,6 +218,60 @@ def got_total_items_first_last_msg(items):
     return f'Got {TOTAL_ITEMS_MARKER} {items}: {FIRST_ITEM_MARKER} - {LAST_ITEM_MARKER}' + '\n'
 
 
+def process_all_pages_result(page, items, all_items, total_items, page_message, message_attribute):
```

Comment:
what are we gaining by pulling this into a function? Also the name vs description is confusing, are we processing all pages or one page here?